core/unix/module_elf.c

```diff
@@ -1575,226 +1553,3 @@ module_relocate_rela(app_pc modbase, os_privmod_data_t *pd, ELF_RELA_TYPE *start
     for (rela = start; rela < end; rela++)
         module_relocate_symbol((ELF_REL_TYPE *)rela, pd, true);
 }
-
-#ifdef LINUX
-/**************************************************************************************
- * Restartable sequence ("rseq") support (i#2350).
- * This is a kernel feature which provides cpu-atomic regions: if a thread
- * is pre-empted within an rseq region, an abort handler is invoked.
- * The feature is difficult to handle under binary instrumentation.
- * We rely on the app following certain conventions, including containing a
- * section holding a table of all rseq sequences.
- */
-
-static void
-rseq_process_entry(struct rseq_cs *entry, ssize_t load_offs)
-{
-    LOG(GLOBAL, LOG_LOADER, 2,
-        "Found rseq region: ver=%u; flags=%u; start=" PFX "; end=" PFX "; abort=" PFX
-        "\n",
-        entry->version, entry->flags, entry->start_ip + load_offs,
-        entry->start_ip + entry->post_commit_offset + load_offs,
-        entry->abort_ip + load_offs);
-    app_pc start_pc = (app_pc)(ptr_uint_t)entry->start_ip + load_offs;
-    vmvector_add(d_r_rseq_areas, start_pc, start_pc + entry->post_commit_offset,
-                 (void *)((ptr_uint_t)entry->abort_ip + load_offs));
-    RSTATS_INC(num_rseq_regions);
-    /* Check the start pc.  We don't take the effort to check for non-tags or
-     * interior pc's.
-     */
-    if (fragment_lookup(GLOBAL_DCONTEXT, start_pc) != NULL) {
```

Comment:
I was just asking the same question and then saw the comment that we're not testing the interior of all fragments... seems reasonable...